### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by ikkala**
Added proposal for eslint configuration:

- For apps, add eslint with baseline .eslintrc.json
  - The "globals" section is based on https://banglejs.com/reference
    - Added `g` and `WIDGETS` to there too because they seemed to be in use in code of apps
    - Instead of using "globals" for all globals, in next phase, the "env" section could define for example "espruino" and "banglejs" for getting all the globals, like proposed in https://github.com/espruino/BangleApps/issues/45#issuecomment-583274072
  - The settings in "rules" section of this baseline .eslintrc.json are selected so that current apps will pass the checks
    - People don't have to immediately go and fix multiple places, or like said [here: "I feel like the interests of being friendly to newcomers and getting a wide variety of apps probably outweighs having them all looking identical."](https://github.com/espruino/BangleApps/issues/45#issuecomment-554951498)
    - As an exception to above, activated "no-mixed-spaces-and-tabs" (implements partly the "Double spaces for indent (no tabs)" from https://github.com/espruino/BangleApps/issues/45#issuecomment-607217736 nicely) and  "no-extra-semi" because they were affecting the current apps so lightly
    - Also, for example setting `"no-undef": "warn"` (currently [commented out](https://github.com/ikkala/BangleApps/blob/593a9240d799788248e9fd4ac5893531be2ec0a1/apps/.eslintrc.json#L147)) would be soft way to nudge people to good direction 
- buffgym already had its own .eslintrc.json, so made the app to honor it:
  - Pass redraw as parameter to Workout, instead of using global variable.
  - Make code to honor eslint linebreak-style (use Windows linebreak).
  - Change eslintrc indent rule to match code style.
  - Add missing semicolons
- npm test: Run eslint after sanitycheck.js
  - Added also separate npm run target: npm run target "lint-apps"

### CODE REVIEW

**Comment by ra100** [comment]
Looks good to me.
Just one question about semis of course :D Would it make sense to not use, to "save" some space? But I think it's not too relevant, and anyway it's safer to have semis there, maybe even better for newcomers.

**Comment by ikkala** [comment]
Saving space argument could also be used in case of tab vs space indentation. But, in general, I guess the space usage should be settled by minimizing the code instead.

And in more general, I think it's good idea to use the defaults that JS community in general use -- or the defaults the standard / defacto standard tools are using. In this case, this means using eslint:recommended as much as possible.  :)

**Comment by ikkala** [comment]
If this pull request can be taken in some form to master I have next on in pipe: https://github.com/ikkala/BangleApps/commit/c6fd71be0333cc317d4b7f36dccd7460f67ccac3
It finalizes the implementation of "Double spaces for indent (no tabs)" of http://www.espruino.com/Code+Style.

**Comment by ra100** [comment]
Yes, I agree, go with the default. And minimizing code is a good point. :)

File: apps/berlinc/berlin-clock.js [hunk 1]
```diff
@@ -88,10 +88,9 @@ Bangle.on('lcdPower', (on) => {
   g.clear();
   if (on) {
     Bangle.drawWidgets();
-    // call your app function here
-    drawBerlinClock();
-  }
-});
+  // call your app function here
+  drawBerlinClock();
```
Comment (ra100) [inline apps/berlinc/berlin-clock.js:92]: this one looks weird, I'd expect:
```js
  if (on) {
    Bangle.drawWidgets();
    // call your app function here
    drawBerlinClock();
  }
});
```

Comment (ikkala) [reply apps/berlinc/berlin-clock.js:92]: Good catch -- I was doing the eslint automatic fix to [earlier version of this file originally](https://github.com/ikkala/BangleApps/commit/0a32e40b1abdf221b79228a8fb36659c18fa0a00#diff-e593f7fcdc3183a0484d87c313186faa) and after that just blindly auto-merged the newer changes from the author before making todays pull request.

I fix fix this.

**Comment by gfwilliams** [comment]
Nice - thanks for this. Looks great!

**Comment by atjn** [comment]
I have made PR's to fix some of the rules that were disabled to support existing apps:

- #3188
- #3187
- #3186

**Pull request merged**